### PR TITLE
Build & publish a latest release certified-astronomer package

### PR DESCRIPTION
**Pull request by kaxil**
closes https://github.com/astronomer/issues/issues/1043

Allows installing the latest version in 1.10.7 series using "pip install astronomer-certified==1.10.7"

Only builds this package for non-dev releases and when a tag is created

### CODE REVIEW

**Comment by kaxil** [comment]
> Do we need a whole separate file for this? Could we instead make it an option to the existing AC setup

Updated

File: astronomer-certified-setup.py [hunk 1]
```diff
@@ -53,29 +53,45 @@ def get_info_from_airflow_wheel(filepath):
...
+    path_to_airflow_wheel = str(sys.argv[-1])
 sys.argv.pop()
 
-version = transform_airflow_version(airflow_version)
+extras, airflow_version = get_info_from_airflow_wheel(path_to_airflow_wheel)
+
+version = transform_airflow_version(airflow_version, build_latest)
+
+if build_latest:
+    required_airflow_version = "apache-airflow~=" + version
```
Comment (ashb) [inline astronomer-certified-setup.py]: What does this evaluate to?

File: astronomer-certified-setup.py [hunk 2]
```diff
@@ -53,29 +53,45 @@ def get_info_from_airflow_wheel(filepath):
...
+def transform_airflow_version(airflow_ver, build_latest_release=False):
 
     parsed = Version(airflow_ver)
 
     ver = '.'.join(str(x) for x in parsed.release)
 
-    if parsed.local:
-        local = parsed._version.local[1]
-        ver += '-' + str(local)
+    if not build_latest_release:
```
Comment (ashb) [inline astronomer-certified-setup.py]: As a general rule: I find the less indented code is, the easier it is to read:

```python
    if build_latest_release:
        return airflow_ver

    parsed = Version(airflow_ver)

    ver = '.'.join(str(x) for x in parsed.release) 

    ...
```

**Comment by ashb** [comment]
> Allows installing the latest version in 1.10.7 series using "pip install astronomer-certified==1.10.7"

Does it do this by re-publishing an wheel with the same name? That won't work I don't think -- pip won't reinstall it/make any network requests if the local version already matches what the specifier.

File: astronomer-certified-setup.py (hunk 1 above)
Comment (kaxil) [reply astronomer-certified-setup.py]: "apache-airflow~=1.10.7"

**Comment by kaxil** [comment]
> > Allows installing the latest version in 1.10.7 series using "pip install astronomer-certified==1.10.7"
> 
> Does it do this by re-publishing an wheel with the same name? That won't work I don't think -- pip won't reinstall it/make any network requests if the local version already matches what the specifier.

Will post a detailed reply with example in a bit.

**Comment by kaxil** [comment]
```bash
#######################################
❯ python setup.py compile_assets sdist --dist-dir dist/apache-airflow/ bdist_wheel --dist-dir dist/apache-airflow/

# Produces 'apache-airflow-1!1.10.7+astro.7'
❯ ll dist/apache-airflow/
-rw-r--r--  1 kaxilnaik  staff   4.4M 26 Mar 23:30 dist/apache-airflow/apache_airflow-1!1.10.7+astro.7-py2.py3-none-any.whl

#######################################

❯ python astronomer-certified-setup.py bdist_wheel --dist-dir dist/astronomer-certified dist/apache-airflow/apache_airflow-*1.10.7*-py2.py3-none-any.whl

# Produces 'astronomer-certified-1.10.7.post7'
❯ ll dist/astronomer-certified
total 72
-rw-r--r--  1 kaxilnaik  staff    36K 26 Mar 23:35 astronomer_certified-1.10.7.post7-py2.py3-none-any.whl

#######################################

❯ python astronomer-certified-setup.py bdist_wheel --dist-dir dist/astronomer-certified dist/apache-airflow/apache_airflow-*-py2.py3-none-any.whl  build-latest

# Produces 'astronomer-certified-1.10.7'
❯ ll dist/astronomer-certified
total 72
-rw-r--r--  1 kaxilnaik  staff    36K 26 Mar 23:30 astronomer_certified-1.10.7-py2.py3-none-any.whl

```

File: astronomer-certified-setup.py (hunk 2 above)
Comment (kaxil) [reply astronomer-certified-setup.py]: Updated

**Comment by kaxil** [comment]
> Does it do this by re-publishing an wheel with the same name? That won't work I don't think -- pip won't reinstall it/make any network requests if the local version already matches what the specifier.

I will try this and let you know

**Comment by kaxil** [comment]
Closing this and added a comment in https://github.com/astronomer/issues/issues/1043#issuecomment-605044819

**Pull request closed without merging**